### DISCUSSION TITLE
What is the source of truth?

### DISCUSSION THREAD

**Discussion by nandorojo** [nandorojo/solito#57] [Ideas]
There is something missing still: what is the source of truth in a Solito app for navigation? The pages folder vs react-navigation linking config presents opportunities to mess things up. It would be great to add file system routing to Expo. They've talked about adding this, but seem short-staffed and want to hire someone to lead the charge.

I've also found `pathpida` to be a really useful package for adding typesafety to routes and query parameters. Based on your Next.js pages, it codegens a set of functions that provide you with the path to a given page. It's compatible with Solito, so I think it's a good utility. In the BeatGig app, I don't navigate with any direct URL strings. Instead, I have them all wrapped with typed functions.

**Comment by wibb36** [comment]
Are you referring to this: https://github.com/EvanBacon/expo-auto-navigation

**Reply by nandorojo** [reply]
yeah, that would be the ideal use case. @evanbacon would know more about its future. it’ll play a big part of the Solito future if it becomes part of the native approach to navigation.

**Reply by wibb36** [reply]
I've been thinking the same way that having a single source of truth for pages is vital for a cross-platform monorepo... I'm currently using the exact same file structure example `[slug].tsx` in the `app` and `next` packages as this would make it easier to implement such a feature in the future.

In a previous project, I only had one file in the web package; `App.tsx` just like you have it in the expo package but that was when using React Navigation's container for the web, I was relying on deep linking for the web by doing the following in next.config :
```tsx
    async rewrites() {
        return [
            // Rewrite everything to use `pages/index`
            {
                source: '/:path*',
                destination: '/',
            },
        ]
    },
```
I'm not sure if that's the right approach when not using RN `NavigationContainer`. Maybe another approach to solving this is by having a custom `NavigationContainer` for the web as well, which will also help to have cross-platform navigators (tabs, drawer, etc) implemented cleaner and easier.

In case you have one source of truth for file system routing, how would you be able to customize pages that you only need on the web(ex: landing page)?
